### ISSUE TITLE
Error attempting to upload findings to Defect Dojo

### ISSUE THREAD

**Issue by kmouzoul23** [DependencyTrack/dependency-track#2628]
### Current Behavior

Cannot use Defect Dojo integration - Defect Dojo Integration error when attempting to upload findings

log output:
ERROR [DefectDojoClient] Make the subsequent pagination call on http:[//]hostname/api/v2/tests/?engagement=1842&limit=100&offset=4500
2023-03-28 09:34:54,227 ERROR [DefectDojoUploader] An error occurred attempting to upload findings to DefectDojo
java.lang.NullPointerException: Cannot invoke "Object.toString()" because the return value of "kong.unirest.json.JSONObject.get(String)" is null
        at org.dependencytrack.integrations.defectdojo.DefectDojoClient.getDojoTestId(DefectDojoClient.java:123)
        at org.dependencytrack.integrations.defectdojo.DefectDojoUploader.upload(DefectDojoUploader.java:90)
        at org.dependencytrack.tasks.VulnerabilityManagementUploadTask.processProjectFindings(VulnerabilityManagementUploadTask.java:66)
        at org.dependencytrack.tasks.VulnerabilityManagementUploadTask.inform(VulnerabilityManagementUploadTask.java:46)
        at org.dependencytrack.tasks.DefectDojoUploadTask.inform(DefectDojoUploadTask.java:37)
        at alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:101)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
2023-03-28 09:34:54,228 ERROR [DefectDojoUploader] An error occurred with the DefectDojo integration point
java.lang.NullPointerException: Cannot invoke "Object.toString()" because the return value of "kong.unirest.json.JSONObject.get(String)" is null
        at org.dependencytrack.integrations.defectdojo.DefectDojoClient.getDojoTestId(DefectDojoClient.java:123)
        at org.dependencytrack.integrations.defectdojo.DefectDojoUploader.upload(DefectDojoUploader.java:90)
        at org.dependencytrack.tasks.VulnerabilityManagementUploadTask.processProjectFindings(VulnerabilityManagementUploadTask.java:66)
        at org.dependencytrack.tasks.VulnerabilityManagementUploadTask.inform(VulnerabilityManagementUploadTask.java:46)
        at org.dependencytrack.tasks.DefectDojoUploadTask.inform(DefectDojoUploadTask.java:37)
        at alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:101)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)

### Steps to Reproduce

1. Enable Defect Dojo integration
2. Integrate projects with Defect Dojo (Create Project Property)

### Expected Behavior

Dependency-Track Findings report should be uploaded successfully to Defect Dojo

### Dependency-Track Version

4.7.1

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Browser

Google Chrome

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by corvusmod** [comment]
As a workaround what I did is to use the FPF format from DT.
I exported the results from DT in FPF format through the API and upload them to DefectDojo

**Comment by SeanWrightFeat** [comment]
I'm getting the same error. Currently using the following:
* Dependency Track: 4.8.0
* DefectDojo: 2.22.0

I noticed the following from the DefectDojo logs:
```
nginx_1         | 172.20.0.1 - - [04/May/2023:20:04:32 +0000] "GET /api/v2/tests/?limit=100&engagement=23 HTTP/1.0" 200 66 "-" "Apache-HttpClient/4.5.14 (Java/17.0.6)" "10.65.32.201"
nginx_1         | 2023/05/04 20:04:32 [warn] 7#7: *157 a client request body is buffered to a temporary file /var/cache/nginx/client_temp/0000000007, client: 172.20.0.1, server: , request: "POST /api/v2/import-scan/ HTTP/1.0", host: "dojo.featurespace.net"
uwsgi_1         | [04/May/2023 20:04:32] WARNING [django.request:241] Bad Request: /api/v2/import-scan/
uwsgi_1         | [pid: 22|app: -|req: -/-] 172.20.0.1 (-) {48 vars in 833 bytes} [Thu May  4 20:04:32 2023] POST /api/v2/import-scan/ => generated 140 bytes in 40 msecs (HTTP/1.0 400) 8 headers in 244 bytes (1 switches on core 0)
nginx_1         | 172.20.0.1 - - [04/May/2023:20:04:32 +0000] "POST /api/v2/import-scan/ HTTP/1.0" 400 140 "-" "Apache-HttpClient/4.5.14 (Java/17.0.6)" "10.65.32.201"
```

**Comment by kmouzoul23** [comment]
It's an error related to pagination. Based on the follwing issue: https://github.com/DependencyTrack/dependency-track/issues/2707 seems that will be fixed in v4.8.1

**Comment by nscuro** [comment]
While it looks related at first, the issue described here is related to when the `scan_type` in `test`s of the DD response are null. Our logic assumed that `scan_type` can not be null, but apparently it can.

<img width="587" alt="image" src="https://github.com/DependencyTrack/dependency-track/assets/5693141/9b96e8f3-6d88-4f76-b30d-d51346bbbd9b">
